examples/lines_bars_and_markers/fill_between_demo.py

```diff
@@ -3,22 +3,27 @@
 Filling the area between lines
 ==============================
 
-This example shows how to use ``fill_between`` to color between lines based on
-user-defined logic.
+This example shows how to use `~.axes.Axes.fill_between` to color the area
+between two lines.
 """
 
 import matplotlib.pyplot as plt
 import numpy as np
 
+###############################################################################
+#
+# Basic usage
+# -----------
+# The parameters *y1* and *y2* can be a scalar, indicating a horizontal
+# boundary a  the given y-values. If only *y1* is given, *y2* defaults to 0.
+
 x = np.arange(0.0, 2, 0.01)
 y1 = np.sin(2 * np.pi * x)
 y2 = 1.2 * np.sin(4 * np.pi * x)
 
-###############################################################################
-
 fig, (ax1, ax2, ax3) = plt.subplots(3, 1, sharex=True)
 
-ax1.fill_between(x, 0, y1)
+ax1.fill_between(x, y1)
```

Comment:
the y labels look really crammed, consider rewriting them to e.g. `from foo to bar` (shorter) or use titles or text(...).

Reply:
Done.